lib/actions/field-trip.js

```diff
@@ -0,0 +1,424 @@
+import moment from 'moment'
+import qs from 'qs'
+import { createAction } from 'redux-actions'
+
+import {routingQuery} from './api'
+import {setQueryParam} from './form'
+import {getGroupSize, getTripFromRequest, sessionIsInvalid} from '../util/call-taker'
+
+if (typeof (fetch) === 'undefined') require('isomorphic-fetch')
+
+/// PRIVATE ACTIONS
+
+const receivedFieldTrips = createAction('RECEIVED_FIELD_TRIPS')
+const requestingFieldTrips = createAction('REQUESTING_FIELD_TRIPS')
+const receivedFieldTripDetails = createAction('RECEIVED_FIELD_TRIP_DETAILS')
+const requestingFieldTripDetails = createAction('REQUESTING_FIELD_TRIP_DETAILS')
+
+// PUBLIC ACTIONS
+
+export const setFieldTripFilter = createAction('SET_FIELD_TRIP_FILTER')
+export const setActiveFieldTrip = createAction('SET_ACTIVE_FIELD_TRIP')
+export const setGroupSize = createAction('SET_GROUP_SIZE')
+export const toggleFieldTrips = createAction('TOGGLE_FIELD_TRIPS')
+
+/**
+ * Fetch all field trip requests (as summaries).
+ */
+export function fetchFieldTrips () {
+  return function (dispatch, getState) {
+    dispatch(requestingFieldTrips())
+    const {callTaker, otp} = getState()
+    if (sessionIsInvalid(callTaker.session)) return
+    const {datastoreUrl} = otp.config
+    const {sessionId} = callTaker.session
+    fetch(`${datastoreUrl}/fieldtrip/getRequestsSummary?${qs.stringify({sessionId})}`)
+      .then(res => res.json())
+      .then(fieldTrips => dispatch(receivedFieldTrips({fieldTrips})))
+      .catch(err => alert(`Error fetching field trips: ${JSON.stringify(err)}`))
+  }
+}
+
+/**
+ * Fetch details for a particular field trip request.
+ */
+export function fetchFieldTripDetails (requestId) {
+  return function (dispatch, getState) {
+    dispatch(requestingFieldTripDetails())
+    const {callTaker, otp} = getState()
+    if (sessionIsInvalid(callTaker.session)) return
+    const {datastoreUrl} = otp.config
+    const {sessionId} = callTaker.session
+    fetch(`${datastoreUrl}/fieldtrip/getRequest?${qs.stringify({requestId, sessionId})}`)
+      .then(res => res.json())
+      .then(fieldTrip => dispatch(receivedFieldTripDetails({fieldTrip})))
+      .catch(err => {
+        alert(`Error fetching field trips: ${JSON.stringify(err)}`)
+      })
+  }
+}
+
+/**
+ * Add note for field trip request.
+ */
+export function addFieldTripNote (request, note) {
+  return function (dispatch, getState) {
+    const {callTaker, otp} = getState()
+    const {datastoreUrl} = otp.config
+    if (sessionIsInvalid(callTaker.session)) return
+    console.log(callTaker.session)
+    const {sessionId, username} = callTaker.session
+    const queryData = new FormData()
+    queryData.append('sessionId', sessionId)
+    queryData.append('note.note', note.note)
+    queryData.append('note.type', note.type)
+    queryData.append('note.userName', username)
+    queryData.append('requestId', request.id)
```

Comment:
This appears enough that it might be useful to use this npm package: https://www.npmjs.com/package/object-to-formdata.

Reply:
Thanks for the suggestion. Added in 6fd2676c6873e54c6810c570238f83c1f8184826.